EpsilonAuxBMS/Src/SetContactorsTask.c

```diff
@@ -47,16 +47,41 @@ void resetSenseArray(uint32_t senses[]);
 // Function for checking if CURRENT_SENSE is at a stable value
 int isSenseStable(uint32_t senses[]);
 
+// Functions for state's logic.
+void firstCheck();
+void commonEnableCheck();
+void chargeEnableCheck();
+void dischargeEnableCheck();
+void done();
+void blocked();
+
+// Global pointers to be used by the state logic functions
+ContactorsSettingState *ptrState;
+ContactorsSettingState *ptrPrevState;
+uint8_t *ptrHasChargeBeenSet;
+uint8_t *ptrHasDischargeBeenSet;
```

Comment:
Instead of making these pointers, you could keep them as global variables and do the one time assignment to them below.

Reply:
I was the one that recommended pointers, but looking at it now I think that probably isn't the best idea :/
What do you think of making make an entire struct collecting these 4 fields together, then you could do like:
```
myStruct A(struct ) {
//blah
return aStruct
}
switch(myStuct.state){
case A:
 myStruct = A(myStruct)
}
```

Reply:
Wooow Bill tsk tsk. Yeah it was a bit messy to have all those asterisks everywhere.
I think what may be better is if the argument to the functions was a pointer to a struct, then they can use the `->` to dereference the values. And it will eliminate the need to keep assigning to the struct and returning a struct in each function. So like:

`void A(struct* aStruct){
// blah
if(aStruct->val) doSomething;
aStruct->state = B;
}

void setContactors(){
  // blah
  struct myStruct;
  // blah
  switch(myStruct.state){
  case A:
    A(&myStruct);
  case B:
    //blah
  }
`